resolve/index_test.go

```diff
@@ -0,0 +1,152 @@
+package resolve_test
+
```

Comment:
Thanks for writing a unit test. Please also write an integration test that makes use of the new flags. I found tests that run Gazelle and check the real output are much better at finding bugs and regressions than unit tests. They're also generally simpler to write since we have the machinery in place already.

Integration tests are in `cmd/gazelle/integration_test.go`. For this, I'd suggest something like:

1. Run with `-experimental_write_index_path=index.json` in a small workspace with .go files for two libraries, one importing the other from different directories.
2. Check that the build files have the expected content.
3. Change the build file for the imported library to have a different label.
4. Run again with `-experimental_read_index_path=index.json` only in the importing library's directory. Nothing should change.
5. Run again without the index. The `deps` for the importing library should change.